src/components/canvas/Stars.js

```diff
@@ -1,251 +1,250 @@
-import React from 'react';
+import React from 'react'; // teacher
 import { connect } from 'react-redux';
 import Star from './Star';
+import { STAR_OPACITY_REDUCTION_FACTOR } from '../../config/settings';
+
+// compute opacity when number of lamp posts increases
+const calculateOpacity = numLampPosts =>
+  numLampPosts * STAR_OPACITY_REDUCTION_FACTOR;
 
 const generateStars = lampPosts => {
   // do some complex logic
   const numLampPosts = lampPosts.length;
   return [
     {
-      x: 100,
-      y: 100,
-      opacity: 1 - numLampPosts * 0.1,
-      r: 1,
-    },
-    {
-      x: 300,
-      y: 150,
-      opacity: 0.7 - numLampPosts * 0.1,
-      r: 1,
+      x: 51,
+      y: 151,
+      opacity: 1 - calculateOpacity(numLampPosts),
```

Comment:
You could potentially factor out the `1 - ` to the `calculateOpacity` function as it always a part of the calculation.